google-cloud-firestore/src/test/java/com/google/cloud/firestore/BulkWriterTest.java

```diff
@@ -422,8 +743,8 @@ public void retriesIndividualWritesThatFailWithAbortedOrUnavailable() throws Exc
       result1.get();
       fail("set() should have failed");
     } catch (Exception e) {
-      assertTrue(e.getCause() instanceof FirestoreException);
-      assertEquals(Status.DEADLINE_EXCEEDED, ((FirestoreException) e.getCause()).getStatus());
+      assertTrue(e.getCause() instanceof BulkWriterError);
```

Comment:
Could `BulkWriterError` extend `FirestoreException`? That way, it can be commonly treated by our users.

`BulkWriterError` should likely also be `BulkWriterException`. See the first part of the first answer here:  https://stackoverflow.com/questions/5813614/what-is-difference-between-errors-and-exceptions

Reply:
I attempted it. I had to make `FirestoreException` not `final` in order to extend, and I had to make one of the private constructors package-private.